### ISSUE TITLE
`Cannot read properties of undefined (reading 'type')` error happens when `manualChunks` is used

### ISSUE THREAD

**Issue by sapphi-red** [rollup/rollup#4908]
### Rollup Version

3.19.1 (from 3.13.0, 3.12.1 is fine)

### Operating System (or Browser)

stackblitz

### Link To Reproduction

https://stackblitz.com/edit/rollup-repro-8wae7q?file=rollup.config.js

### Expected Behaviour

No error happens.

### Actual Behaviour

The following error happens when building:

```
❯ npm run build
$ rollup -c rollup.config.js

src/main.js → dist...
[!] TypeError: Cannot read properties of undefined (reading 'type')
    at FunctionExpression.deoptimizeArgumentsOnInteractionAtPath (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:9075:25)
    at FunctionExpression.deoptimizeArgumentsOnInteractionAtPath (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:9471:15)
    at CallExpression.applyDeoptimizations (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:10192:21)
    at CallExpression.hasEffects (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:10158:22)
    at ExpressionStatement.hasEffects (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:6268:28)
    at Program.hasEffects (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:12133:22)
    at Program.hasCachedEffects (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:12128:44)
    at Module.hasEffects (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:14019:75)
    at addSideEffectDependencies (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:14359:72)
    at Module.addRelevantSideEffectDependencies (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:14366:9)
```

rollup 3.13.0 throws a different error:
```
[!] TypeError: Cannot read properties of undefined (reading 'thisArg')
    at CallExpression.applyDeoptimizations (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:10025:30)
    at CallExpression.hasEffects (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:9991:22)
    at ExpressionStatement.hasEffects (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:6252:28)
    at Program.hasEffects (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:11949:22)
    at Program.hasCachedEffects (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:11944:44)
    at Module.hasEffects (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:13845:75)
    at addSideEffectDependencies (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:14185:72)
    at Module.addRelevantSideEffectDependencies (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:14192:9)
    at Module.getDependenciesToBeIncluded (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:13706:18)
    at addStaticDependencies (file:///home/projects/rollup-repro-8wae7q/node_modules/rollup/dist/shared/rollup.js:15413:33)
```

Original issue: https://github.com/vitejs/vite/issues/12448

**Comment by timdang** [comment]
For us, migrating from Vite 4.1.4 to 4.2.1 bumped our Rollup dependency to 3.20.0 and we're no experiencing this issue as well.

**Comment by TrickyPi** [comment]
It seems that the package in `manualChunks` config of  Rollup@latest  must included in  module graph

**Comment by lukastaegert** [comment]
> The earliest version of Rollup with this bug is 3.13.0

Thanks @TrickyPi, that was extremely helpful! It turns out, 3.13.0 only contained one seemingly unrelated feature. But in implementing the feature I accidentally removed a conditional that was important to handle the case where the entry of a manual chunk is not part of the module graph. Fix at #4921
